build_matrix_A.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+
+
+def build_matrix_A(material_property_library, mesh):
+    """ This function will take a previously generated mesh and create the 'A'
+    matrix used in solving this problem.
+
+    inputs:
+    --------
+    - material_property_library: this will be a dictionary or nested dictionary
+    containing all of the important data for each material (fuel,
+    various layers of graphite)
+    - mesh: the physical location of each mesh point
+    - materials: the materials in each region and how many nodes are assigned to that material
+    outputs. List of tuples of format (material, number of nodes)
+    --------
+    - An array that can be plugged into a subsequent function to solve the
+    A.x=b formula
+    """
+    k = np.zeros(len(mesh))
```

Comment:
Is `len(mesh)` just the number of material layers or is it the number of discretized points?

Reply:
len(mesh) is the number of materials used, so this loop makes lists of material properties with an entry for each material.